sklearn/ensemble/gradient_boosting.py

```diff
@@ -1417,11 +1420,20 @@ def fit(self, X, y, sample_weight=None, monitor=None):
             # init state
             self._init_state()
 
-            # fit initial model - FIXME make sample_weight optional
-            self.init_.fit(X, y, sample_weight)
+            # fit initial model
+            if has_fit_parameter(self.init_, "sample_weight"):
```

Comment:
We should avoid has_fit_parameter unless try-except really won't work.

Reply:
It means keyword args won't work. We should only use this kind of thing when we really need to take a different approach when sample_weight is not supported. (Feel free to document this caveat more in a separate pr)